packages/core/src/shorthands/flex.test.ts

```diff
@@ -0,0 +1,99 @@
+import { flex } from './flex';
+
+describe('flex', () => {
+  it('for a given value auto', () => {
+    expect(flex('auto')).toEqual({
+      flexGrow: 1,
+      flexShrink: 1,
+      flexBasis: 'auto',
+    });
+  });
+
+  it('for a given value initial', () => {
+    expect(flex('initial')).toEqual({
+      flexGrow: 0,
+      flexShrink: 1,
+      flexBasis: 'auto',
+    });
+  });
+
+  it('for a given value none', () => {
+    expect(flex('none')).toEqual({
+      flexGrow: 0,
+      flexShrink: 0,
+      flexBasis: 'auto',
+    });
+  });
+
+  it('for a given value 2', () => {
+    expect(flex(2)).toEqual({
+      flexGrow: 2,
+      flexShrink: 1,
+      flexBasis: 0,
+    });
+  });
+
+  it('for a given value 10em', () => {
+    expect(flex('10em')).toEqual({
+      flexGrow: 1,
+      flexShrink: 1,
+      flexBasis: '10em',
+    });
+  });
+
+  it('for a given value 30%', () => {
+    expect(flex('30%')).toEqual({
+      flexGrow: 1,
+      flexShrink: 1,
+      flexBasis: '30%',
+    });
+  });
+
+  it('for a given value min-content', () => {
+    expect(flex('min-content')).toEqual({
+      flexGrow: 1,
+      flexShrink: 1,
+      flexBasis: 'min-content',
+    });
+  });
+
+  it('for a given values 1, 30px', () => {
+    expect(flex(1, '30px')).toEqual({
+      flexGrow: 1,
+      flexShrink: 1,
+      flexBasis: '30px',
+    });
+  });
+
+  it('for a given values 2, 2', () => {
+    expect(flex(2, 2)).toEqual({
+      flexGrow: 2,
+      flexShrink: 2,
+      flexBasis: 'auto',
+    });
+  });
```

Comment:
```suggestion
  it('for a given values 2, 2', () => {
    expect(flex(2, 2)).toEqual({
      flexGrow: 2,
      flexShrink: 2,
      flexBasis: 0,
    });
  });
```

[The spec](https://drafts.csswg.org/css-flexbox/#flex-common) is not really helpful there, but it seems that it should be `flexBasis: 0` in this case (the same behavior as in `flex: 2`):

![image](https://user-images.githubusercontent.com/14183168/172136705-7ce5a288-7629-4c56-82f6-262a60aea131.png)

![image](https://user-images.githubusercontent.com/14183168/172136831-6cf59495-123d-4c04-a1ae-17fa143c538c.png)

Please correct me if I am wrong.

Reply:
Oh interesting… I was applying `auto`, because it is the initial value for `flex-basis`. But you are absolutely right! Will fix!

> ''flex: [<number [1,∞]>](https://drafts.csswg.org/css-values-4/#number-value)''
Equivalent to ''flex: [<number [1,∞]>](https://drafts.csswg.org/css-values-4/#number-value) 1 0''. Makes the flex item flexible and **sets the [flex basis](https://drafts.csswg.org/css-flexbox/#flex-flex-basis) to zero**, resulting in an item that receives the specified proportion of the free space in the flex container. If all items in the flex container use this pattern, their sizes will be proportional to the specified flex factor.

— [Flex property](https://drafts.csswg.org/css-flexbox/#flex-property)